### DISCUSSION TITLE
Generating forms by reading the Wiki Mapping Table

### DISCUSSION THREAD

**Discussion by MusaabKh** [SmartDataAnalytics/OpenResearch#132] [Ideas]
I was looking into it and saw the form code can be generated using a generator in Python. The idea is to use wikiquery to query the page and then format it accordingly to the form code and push it to the wiki using wikirestore from the py-3rd-partymediawiki package. 
Also I had another question that for the wikirestore is the source code for the form only required or do we need to add some extra things to make it a wiki file?

**Comment by WolfgangFahl** [comment]
Templates and Forms are regular wiki markup files. If you can generate the ".wiki" files you are basically all set if you'd have a wikirestore for the "installation". The advantages is that you'd be able to test things in different environments if you pursue this

**Comment by WolfgangFahl** [comment]
Of course you need to "bootstrap" your environment with a reasonable set of templates and forms to get your environment working. That's why we were using wikpush initially to make available the Templates and Forms for the MetaModel elements that let you allow to e.g. handle Properties and Topics and or Concepts.
